### PR TITLE
Implement globbing (pathname expansion)

### PR DESCRIPTION
**Pull request by jbrinksma**
## Description:

Globbing AKA pathname expansion

**Related issue (if applicable):** fixes #<issue number goes here>

## Checklist:
  - [x] The code change works
  - [x] Passes all tests: `make test`
  - [x] There is no commented out code in this PR.
  - [x] `norminette srcs libft | grep -E "^Error" | wc -l` is not higher than master. If it is, run `norminette srcs libft | grep -E "^Error" -B 1` to see errors
  - [x] I solemny swear my code is compliant with the [README][readme-file]

[readme-file]: https://github.com/OscarMulder/codam-42sh/blob/master/README.md

### CODE REVIEW

File: srcs/globbing/glob_tokenlst_funcs.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	glob_deltoken(token);
+}
+
+t_globtoken	*glob_tokenlstnew(char *word_chunk, int type)
+{
+	t_globtoken	*globtoken;
+
+	if (word_chunk == NULL)
+		return (NULL);
+	globtoken = ft_memalloc(sizeof(t_globtoken));
```
Comment (thijsdejong) [inline srcs/globbing/glob_tokenlst_funcs.c]: implicit cast

File: srcs/globbing/glob_tokenlst_funcs.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/*   Updated: 2019/10/21 13:29:50 by mavan-he      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "vsh.h"
+
+void		glob_deltoken(t_globtoken **token)
+{
+	ft_strdel(&(*token)->word_chunk);
+	free(*token);
```
Comment (thijsdejong) [inline srcs/globbing/glob_tokenlst_funcs.c]: You use free here, but on line 42 you use memdel

File: srcs/globbing/glob_matchlst_funcs.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		return ;
+	glob_del_matchlst(&(*matchlst)->next);
+	glob_delmatch(matchlst);
+}
+
+t_globmatchlst	*glob_matchlstnew(char *item)
+{
+	t_globmatchlst *new;
+
+	new = ft_memalloc(sizeof(t_globmatchlst));
```
Comment (thijsdejong) [inline srcs/globbing/glob_matchlst_funcs.c]: implicit cast

File: srcs/globbing/glob_matchlst_funcs.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+/*   Updated: 2019/10/21 13:27:34 by mavan-he      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "vsh.h"
+
+void			glob_delmatch(t_globmatchlst **match)
+{
+	ft_strdel(&(*match)->word);
+	free(*match);
```
Comment (thijsdejong) [inline srcs/globbing/glob_matchlst_funcs.c]: you use free here, but why not use memdel if you're using strdel one line above

File: srcs/globbing/glob_lexer.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		return (err_ret_exit(E_ALLOC_STR, EXIT_FAILURE));
+	glob_tokenlstadd(lst, new);
+	return (FUNCT_SUCCESS);
+}
+
+int				glob_lexer(t_globtoken **lst, char *word)
+{
+	t_globscanner		*scanner;
+
+	scanner = ft_memalloc(sizeof(t_globscanner));
```
Comment (thijsdejong) [inline srcs/globbing/glob_lexer.c]: implicit cast

File: srcs/expan/expan_pathname.c [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+{
+	t_ast *to_free;
+
+	ast->flags = 0;
+	ft_strdel(&ast->value);
+	ast->value = (*expanded)->value;
+	ast->left = (*expanded)->left;
+	to_free = *expanded;
+	*expanded = (*expanded)->left;
+	free(to_free);
```
Comment (thijsdejong) [inline srcs/expan/expan_pathname.c]: use ft_memdel if you're also using strdel

**Comment by jbrinksma** [comment]
Please fix exit_code bug (e.g. 'cat ft_*' will give exit code 1 when there are matches)

**Pull request merged**